src/main/kotlin/com/hibob/academy/resource/ex10-controler.kt

```diff
@@ -0,0 +1,72 @@
+package com.hibob.academy.resource
+
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.core.Response.Status
+import java.time.LocalDate
+
+@Path("/pets")
+class PetController {
+
+    // GET request to retrieve pet type by ID
+    @GET
+    @Path("/{petId}/{type}")
+    @Produces("application/json")
+    fun getPetType(@PathParam("petId") petId: Long): Response {
+        return if (petId < 0) {
+            Response.status(Status.NOT_FOUND).entity("No pet found").build()
+        } else {
+            Response.ok("parrot").build()
+        }
+    }
+
+    // POST request to create a new pet
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    fun createPet(pet: Pet): Response {
+        // Simulate creation logic
+        return Response.status(Status.CREATED).entity("Pet ${pet.name} created successfully").build()
+    }
+
+    // PUT request to update an existing pet
+    @PUT
+    @Path("/{petId}")
+    @Consumes("application/json")
+    @Produces("application/json")
+    fun updatePet(@PathParam("petId") petId: Long, pet: Pet): Response {
+        return if (petId < 0) {
+            Response.status(Status.NOT_FOUND).entity("No pet found to update").build()
+        } else {
+            Response.ok("Pet ${petId} updated successfully").build()
+        }
+    }
+
+    // DELETE request to delete a pet by ID
+    @DELETE
+    @Path("/{petId}")
+    @Produces("application/json")
+    fun deletePet(@PathParam("petId") petId: Long): Response {
+        return if (petId < 0) {
+            Response.status(Status.NOT_FOUND).entity("No pet found to delete").build()
+        } else {
+            Response.ok("Pet ${petId} deleted successfully").build()
+        }
+    }
+
+    // Example endpoint to throw an exception
+    @GET
+    @Path("/error")
+    @Produces("application/json")
+    fun throwError(): Response {
```

Comment:
Nice. You could have also thrown a NotFoundException in the other calls instead of creating the Response.status(Status.NOT_FOUND). It's not wrong but sometimes you would prefer to throw exceptions. 
Most applications will catch these exceptions before returning the status and log them somewhere. This helps the dev's to see the flow and possibly log more data to understand what happened.

So in once of your cases `throw NotFoundException("No pet with $id found to delete")`